### PR TITLE
feat: add take a photo button in profile

### LINKED ISSUE #5518: Feat: capture new image option while editing profile picture
### Description:
While editing profile, user can change his/her profile pic via image upload, album or avatar URL.
An option to capture a new profile picture can be added.

### Environment Information:

- Rocket.Chat Server Version: 6.6.0
- Rocket.Chat App Version: 4.46.0.56378
- Device Name: Realme Narzo 50
- OS Version: Realme UI 4.0

### Steps to reproduce:

1. Go to profile
2. Press on edit

### Expected behavior:

User should have option to able to click a new picture also.

### Actual behavior:
there is only upload, select from album and URL options available

WhatsApp Image 2024-01-28 at 11 57 39

### PR DESCRIPTION
**Pull request by Amitlpu-840**
## Proposed changes

Adds functionality to capture a new photo using mobile camera while updating Avatar
## Issue(s)	

closes #5518 
## How to test or reproduce

- Go to profile
- Press edit 
## Screenshots
WhatsApp Image 2024-01-30 at 15 10 50

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Amitlpu-840** [comment]
please let me know if changes are need.

File: app/views/ChangeAvatarView/index.tsx [hunk 1]
```diff
@@ -130,18 +130,19 @@ const ChangeAvatarView = () => {
 		return navigation.goBack();
 	};
 
-	const pickImage = async () => {
+	const pickImage = async (isCam: boolean) => {
```
Comment (GleidsonDaniel) [inline app/views/ChangeAvatarView/index.tsx:133]:
```suggestion
	const pickImage = async (isCam = false) => {
```

File: app/views/ChangeAvatarView/index.tsx [hunk 2]
```diff
@@ -217,12 +218,20 @@ const ChangeAvatarView = () => {
...
+						onPress={() => pickImage(true)}
+						testID='change-avatar-view-upload-image'
+					/>
 					<Button
 						title={I18n.t('Upload_image')}
 						type='secondary'
 						disabled={saving}
 						backgroundColor={colors.editAndUploadButtonAvatar}
-						onPress={pickImage}
+						onPress={() => pickImage(false)}
```
Comment (GleidsonDaniel) [inline app/views/ChangeAvatarView/index.tsx:234]:
```suggestion
						onPress={pickImage}
```

File: app/views/ChangeAvatarView/index.tsx [hunk 3]
```diff
@@ -130,18 +130,19 @@ const ChangeAvatarView = () => {
...
 		const options = {
 			cropping: true,
 			compressImageQuality: 0.8,
 			freeStyleCropEnabled: true,
 			cropperAvoidEmptySpaceAroundImage: false,
 			cropperChooseText: I18n.t('Choose'),
 			cropperCancelText: I18n.t('Cancel'),
-			includeBase64: true
+			includeBase64: true,
+			useFrontCamera: true
```
Comment (GleidsonDaniel) [inline app/views/ChangeAvatarView/index.tsx:142]:
```suggestion
			useFrontCamera: isCam
```

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

File: app/views/ChangeAvatarView/index.tsx (hunk 3 above)
Comment (Amitlpu-840) [reply app/views/ChangeAvatarView/index.tsx]: Done with the changes. thanks for the valuable insights and feedback. please let me know if any further improvements are needed.

**Comment by Amitlpu-840** [comment]
Done with the changes. thanks for the valuable insights and feedback. please let me know if any further improvements are needed.

File: app/views/ChangeAvatarView/index.tsx [hunk 4]
```diff
@@ -217,6 +218,14 @@ const ChangeAvatarView = () => {
 							}
 						/>
 					) : null}
+					<Button
+						title={I18n.t('Take_a_photo')}
+						type='secondary'
+						disabled={saving}
+						backgroundColor={colors.editAndUploadButtonAvatar}
+						onPress={() => pickImage(true)}
+						testID='change-avatar-view-upload-image'
```
Comment (GleidsonDaniel) [inline app/views/ChangeAvatarView/index.tsx:227]:
```suggestion
						testID='change-avatar-view-take-a-photo'
```
The testId is conflicting with the other

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

File: app/views/ChangeAvatarView/index.tsx (hunk 4 above)
Comment (Amitlpu-840) [reply app/views/ChangeAvatarView/index.tsx:227]: I have a doubt,I think there is no test defined for test id change-avatar-view-take-a-photo. so should I add that too?

Comment (GleidsonDaniel) [reply app/views/ChangeAvatarView/index.tsx:227]: If you can, however it is not mandatory.

**Comment by Amitlpu-840** [comment]
in 'images uploaded' section , I noticed only one one image is received as response from getAvatarSuggestion() even after uploading other pic. is there any specific reason or it's a bug?

**Review by GleidsonDaniel — APPROVED** [review]

**Pull request merged**